Changed components to use Carbon grid instead of inlineDiv

peach280 opened the pull request:
# Pull Requests Requirements

- [X] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [X] The PR title follows conventional commit label standards.
- [X] The changes confirm to the OpenElis Global x3 Styleguide and design
      documentation.
- [X] The changes include tests or are validated by existing tests.
- [X] I have read and agree to the Contributing Guidelines of this project.

## Summary

Fixes #978 
Almost all components have been changes to use carbon grid instead of inlineDiv

## Screenshots

The display is same as before but more responsive.

peach280 committed:
Changed components to use Carbon grid instead of inlineDiv

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Merge branch 'develop' into Carbon

peach280 committed:
Changed components to use Carbon grid instead of inlineDiv

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Merge branch 'Carbon' of https://github.com/peach280/OpenELIS-Global-2 into Carbon

peach280 committed:
Changed components to use Carbon grid instead of inlineDiv

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Changed components to use Carbon grid instead of inlineDiv

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Changed components to use Carbon grid instead of inlineDiv

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Changed components to use Carbon grid instead of inlineDiv

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Changed components to use Carbon grid instead of inlineDiv

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Changed components to use Carbon grid instead of inlineDiv

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Changed components to use Carbon grid instead of inlineDiv

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

mozzy11 commented on frontend/src/components/addOrder/SampleType.js:
```diff
@@ -486,40 +488,40 @@ const SampleType = (props) => {
+              onChange={(date) => handleCollectionDate(date)}
+              value={sampleXml.collectionDate}
+              labelText={intl.formatMessage({ id: "sample.collection.date" })}
+              className="inputText"
```
can you get rid of this class , and we purly rely on carbon grid for the layout?

peach280 replied:
I will change it surely.

mozzy11 commented on frontend/src/components/addOrder/SampleType.js:
```diff
@@ -486,40 +488,40 @@ const SampleType = (props) => {
+              }
+              onChange={(time) => handleCollectionTime(time)}
+              value={sampleXml.collectionTime}
+              className="inputText"
```
get rid of the css class here

mozzy11 committed:
Merge branch 'develop' into Carbon

mozzy11 commented:
Thanks @peach280 . 
you need to attach screen shots of all the affected pages to ensure we dont break the layout

peach280 committed:
Changed components to use Carbon grid instead of inlineDiv

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Merge branch 'Carbon' of https://github.com/peach280/OpenELIS-Global-2 into Carbon

peach280 committed:
Merge branch 'develop' into Carbon

peach280 commented:
![pathology](https://github.com/user-attachments/assets/818b035f-1707-4a49-a8e2-de1112b8907a)
![immuno](https://github.com/user-attachments/assets/5b715438-3caf-4c54-b73f-128eded12833)
![reflexrule](https://github.com/user-attachments/assets/fa09229e-5761-4290-8739-051b3976b041)
![patientMenu](https://github.com/user-attachments/assets/3651337c-435c-4060-9db2-c370f7b4fce5)
![billing](https://github.com/user-attachments/assets/29a45113-74ce-4f78-ba88-9b97e0fd3135)
![sampleType](https://github.com/user-attachments/assets/3d58d9aa-db33-42fe-8ab1-0247a294a471)

peach280 committed:
Changed components to use Carbon grid instead of inlineDiv

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Merge branch 'Carbon' of https://github.com/peach280/OpenELIS-Global-2 into Carbon

mozzy11 committed:
Merge branch 'develop' into Carbon

peach280 committed:
Merge branch 'develop' into Carbon

peach280 committed:
Merge branch 'develop' into Carbon

mozzy11 committed:
Merge branch 'develop' into Carbon

mozzy11 commented on frontend/src/components/addOrder/SampleType.js:
```diff
@@ -486,40 +488,37 @@ const SampleType = (props) => {
-            disallowFutureDate={true}
-          />
+        <Grid>
+          <Column style={{ margin: 0 }} lg={16} md={8} sm={4}>
```
did you notice you changed the layout of the componenets here ??

peach280 replied:
Yes I have fixed the layout now.

mozzy11 commented on frontend/src/components/addOrder/OrderEntryAdditionalQuestions.js:
```diff
@@ -48,9 +49,9 @@ export const ProgramSelect = ({
-        {programs.length > 0 && (
-          <div className="inputText">
+      <Grid>
+        <Column lg={8} md={4} sm={4}>
```
you also have a broken layout here

peach280 replied:
Fixed it

mozzy11 commented:
Hello @peach280 . Thanks for the PR.
Please ensure you dont break the existing Layout.

Ensure to add the screen shots before and after for each page you have made changes into

peach280 commented:
I am taking 'before' and 'after' screenshots. I will complete the process for all modified pages and add them to the PR description.

peach280 committed:
Updated

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Merge branch 'Carbon' of https://github.com/peach280/OpenELIS-Global-2 into Carbon

peach280 committed:
Merge branch 'develop' into Carbon

peach280 commented:
I have made the changes to ensure that the layout doesn't break.

peach280 commented:
Here are the before and after images:
I have observed that the layout has changed a little bit in some pages. I will be ready to make any corrections if needed.

**1) Add Sample** 
Before:
![addsamplebefore](https://github.com/user-attachments/assets/398fd28d-2ca6-4cb1-be39-4e925564d321)
After:
![addorderafter](https://github.com/user-attachments/assets/274b6ef2-0ddf-4809-b048-bd94778acdad)



**2)Additional questions**
Before:
![additionalquestionsbefore](https://github.com/user-attachments/assets/de8b0e2b-88ea-4d44-a14f-b47088c8550d)
After:
![additionalquestionsafter](https://github.com/user-attachments/assets/e30bb6a7-7b67-462d-8d7c-e39da11c3b2c)

**3)Billing** 
Before:
![billingbefore](https://github.com/user-attachments/assets/3fe2d538-ed4a-4875-87d8-ca526bf93914)
After:
![billingafter](https://github.com/user-attachments/assets/a2573190-6f1e-4336-a3a8-fbdd7e04bfda)

**4)Menu Utils** 
Before:

![globalmenubefore](https://github.com/user-attachments/assets/05d08457-e0ec-4d42-9859-19ad6e80bb54)

![patientMenubefore](https://github.com/user-attachments/assets/502a9b07-7548-40db-8c05-95094e29bda1)
![nonconformitybefore](https://github.com/user-attachments/assets/bcdda383-b30b-4a77-ae48-24f97dc1e5d5)


After:

![globalmenuafter](https://github.com/user-attachments/assets/0b6782c6-9851-4f51-b041-86056b6e944d)
![nonconformityafter](https://github.com/user-attachments/assets/ab120bed-22cb-4527-a0db-ec81091c1966)
![patientMenuafter](https://github.com/user-attachments/assets/99ed98db-1f20-4d39-8364-03c97e83986a)
![studymenuafter](https://github.com/user-attachments/assets/1592f89a-2d82-4ffd-ad1c-c320e9266152)

**5)Reflex** 
Before:
![reflexrulebefore](https://github.com/user-attachments/assets/f95e8195-e21c-4ab5-b9f0-c2beb4fe8720)
After:
![reflexruleafter](https://github.com/user-attachments/assets/dfe62a3f-4ee5-46d3-aa8f-8a2f6044629e)



**6)Immunohistochemistry**
Before:
![immunobefore](https://github.com/user-attachments/assets/f4aaea7e-9a34-4aae-a97f-a1cc69ae687a)
After:
![immunoafter](https://github.com/user-attachments/assets/21099cf5-6ec6-4ab6-9861-d38b19073bb3)

**7)Modify order**
Before:
![modifyorderbefore](https://github.com/user-attachments/assets/a34759d5-c6cb-434c-ad4c-a291d3bb7b21)
After:
![modifyorderafter](https://github.com/user-attachments/assets/72e6e98b-a159-4095-942f-b83e10478af8)

**8)Pathology**
Before:
![pathologybefore](https://github.com/user-attachments/assets/164ef467-5296-424e-991e-1c905e1d972c)
After:
![pathologyafter](https://github.com/user-attachments/assets/1559c4fd-d58a-467c-a20a-30144d2cd908)

peach280 committed:
Fixed the layout

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Merge branch 'Carbon' of https://github.com/peach280/OpenELIS-Global-2 into Carbon

peach280 commented:
Hi @mozzy11, In immunohistochemistry and pathology pages, after these changes Column by default arranges elements like Filters,My Cases vertically so should I change that?

peach280 committed:
Fixed the layout

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Merge branch 'develop' into Carbon

peach280 committed:
Merge branch 'develop' into Carbon

mozzy11 commented:
You  should rightly use the Coloum element.
Columns can arrage components vertivaly or horizontaly when righly used

ie compoenets below would be arragened horizontal. see [carbon Grid](https://carbondesignsystem.com/elements/2x-grid/overview/)

```
<Grid> 
     <Column lg={8}>
     </Column>
     <Columnlg={8}>
     </Column>
 </Grid>
```

mozzy11 commented on frontend/src/components/addOrder/AddSample.js:
```diff
@@ -106,45 +107,44 @@ const AddSample = (props) => {
-        </div>
-      </Stack>
+      <Grid>
+        <Column lg={16} md={8} sm={4}>
```
this works fine on Add oder but breaks layout on Edit order
![Uploading image.png…]()

mozzy11 commented on frontend/src/components/addOrder/OrderEntryAdditionalQuestions.js:
```diff
@@ -48,9 +49,9 @@ export const ProgramSelect = ({
-      <div className="formInlineDiv">
-        {programs.length > 0 && (
-          <div className="inputText">
+      <Grid fullWidth={true}>
```
The layout is also broken on Edit order but works fine on Add Odrer

![Uploading image.png…]()

peach280 commented:
Thanks @mozzy11  I will make the changes soon and commit them

peach280 committed:
Updated pathology and immuno pages

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Fixed the layout

Signed-off-by: Vaishnavi Bhandari <vaishnavibhandari.128@gmail.com>

peach280 committed:
Merge branch 'Carbon' of https://github.com/peach280/OpenELIS-Global-2 into Carbon

peach280 commented:
Hi @mozzy11, I have corrected the layout of immunohistochemistry and pathology pages and I have also fixed the layout of modify order for both additional questions and add order. 
Here are the updated screenshots:

![editordera](https://github.com/user-attachments/assets/b2d19b4e-fa0a-4a28-86c0-a40908f3c026)
![editorderb](https://github.com/user-attachments/assets/ba7f4fd0-d7d8-48c5-b14c-bbc9a796d92a)

![pathologyafter](https://github.com/user-attachments/assets/9ec85b55-0581-4172-b1ae-ff7bdb77f950)

![immunoafter](https://github.com/user-attachments/assets/303be568-6280-4253-8d9d-6f0992f8fd84)

peach280 committed:
Merge branch 'develop' into Carbon

peach280 committed:
Merge branch 'develop' into Carbon

peach280 committed:
Merge branch 'develop' into Carbon

peach280 commented:
Hi @mozzy11 , are there any changes that I can make to improve this PR? Please provide your feedback.

mozzy11 commented:
Hello @peach280 , this issue is really impontnant , but ive realised it wont work when you do all the changes at ago in the same PR.
Now lets break this into smaller PRs, youll handle one component at ago in a separate PR , this will ensure no change breaks anything

mozzy11 commented:
Youll make one PR at ago with the relevant screenshot of all the ffected pages

peach280 commented:
Ok I will change components one by one and attach screenshots for both large and small screens. Thanks.

peach280 commented:
Hi @mozzy11 , I have made the changes to some components in seperate PRs. Please review and provide your feedback, so that I can continue with all the components

mozzy11 commented:
Thanks @peach280 .
Ill llok at the smaller PRs broken down form this big one
